lib/include/pl/core/preprocessor.hpp

```diff
@@ -23,70 +24,71 @@ namespace pl::core {
 
         ~Preprocessor() override = default;
 
-        hlp::CompileResult<std::string> preprocess(PatternLanguage *runtime, api::Source* source, bool initialRun = true);
+        hlp::CompileResult<std::vector<pl::core::Token>> preprocess(PatternLanguage *runtime, api::Source* source, bool initialRun = true);
 
         void addDefine(const std::string &name, const std::string &value = "");
         void addPragmaHandler(const std::string &pragmaType, const api::PragmaHandler &handler);
         void addDirectiveHandler(const std::string &directiveType, const api::DirectiveHandler &handler);
         void removePragmaHandler(const std::string &pragmaType);
         void removeDirectiveHandler(const std::string &directiveType);
 
-        std::optional<std::string> getDirectiveValue(bool allowWhitespace = false);
+        std::optional<pl::core::Token> getDirectiveValue(unsigned line);
```

Comment:
Generally use sized types like `u32` instead of `unsigned` or `int` and co.

Reply:
I changed all the unsigned types to u32. Let me know if I missed any.